processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -560,23 +562,21 @@ private Accessor getAdderForType(Type collectionProperty, String pluralPropertyN
         if ( collectionProperty.isCollectionType ) {
 
             // this is a collection, so this can be done always
-            if ( !collectionProperty.getTypeParameters().isEmpty() ) {
-                // there's only one type arg to a collection
-                TypeMirror typeArg = collectionProperty.getTypeParameters().get( 0 ).getTypeBound().getTypeMirror();
-                // now, look for a method that
-                // 1) starts with add,
-                // 2) and has typeArg as one and only arg
-                List<Accessor> adderList = getAdders();
-                for ( Accessor adder : adderList ) {
-                    ExecutableElement executable = adder.getExecutable();
-                    if ( executable == null ) {
-                        // it should not be null, but to be safe
-                        continue;
-                    }
-                    VariableElement arg = executable.getParameters().get( 0 );
-                    if ( typeUtils.isSameType( arg.asType(), typeArg ) ) {
-                        candidates.add( adder );
-                    }
+            TypeMirror typeArg = first( collectionProperty.determineTypeArguments( Iterable.class ) ).getTypeBound()
```

Comment:
nice.. I think that the original solution predates the `determineTypeArguments`

Reply:
I think so too 😄